core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java

```diff
@@ -207,10 +205,15 @@ private String getUserFromSignature() throws S3Exception {
   /**
    * Gets the user from the authorization header string for AWS Signature Version 4.
    * @param authorization the authorization header string
+   * @param conf the {@link AlluxioConfiguration} Alluxio conf
    * @return the user
    */
   @VisibleForTesting
-  public static String getUserFromAuthorization(String authorization) throws S3Exception {
+  public static String getUserFromAuthorization(String authorization, AlluxioConfiguration conf)
+      throws S3Exception {
+    if (conf.get(PropertyKey.SECURITY_AUTHENTICATION_TYPE) == AuthType.NOSASL) {
+      return null;
+    }
```

Comment:
Migrated the property key check within this method since it is being called from `CompleteMultipartUploadHandler` directly.

Ideally it should be calling the higher-level `getUser()`, but `getUserFromSignature()` depends on some instanced information at the moment.